src/Util/UtilTrait.php

```diff
@@ -39,4 +87,74 @@ protected function getPhinxTable(?string $plugin = null): string
 
         return $plugin . $table;
     }
+
+    /**
+     * Detect if any legacy phinxlog tables exist in the database.
+     *
+     * Results are cached per connection to avoid repeated queries.
+     *
+     * @param \Cake\Database\Connection $connection Database connection
+     * @return bool True if legacy tables exist
+     */
+    protected function detectLegacyTables(Connection $connection): bool
+    {
+        $cacheKey = $connection->configName();
+
+        if (isset(self::$legacyTablesCache[$cacheKey])) {
+            return self::$legacyTablesCache[$cacheKey];
+        }
```

Comment:
Outside of tests does this actually get hit often? It looks like maybe once a process as the table name is calculated. I think we can skip the complexity of a cache.